test/jdk/javax/swing/JFileChooser/FileSizeCheck.java

```diff
@@ -21,95 +21,232 @@
+    private enum FileSize {
+        F0(    0, "0.0 KB"),
+        F1(    1, "0.1 KB"),
 
-        int[] tempFilesSize = {0, 1, 160, 299, 900, 901, 999_000, 1_000_000, 2_800_000};
+        F99(  99, "0.1 KB"),
+        F100(100, "0.1 KB"),
+        F101(101, "0.2 KB"),
+        F149(149, "0.2 KB"),
+        F150(150, "0.2 KB"),
+        F151(151, "0.2 KB"),
+        F900(900, "0.9 KB"),
+        F901(901, "1.0 KB"),
 
-        tempFilePaths = new Path[tempFilesName.length];
-        PassFailJFrame.addTestWindow(frame);
-        PassFailJFrame.positionTestWindow(frame, PassFailJFrame.Position.HORIZONTAL);
+        F999_000(999_000, "999.0 KB"),
+        F999_001(999_001, "999.1 KB"),
+        F999_900(999_900, "999.9 KB"),
+        F999_901(999_901,   "1.0 MB"),
 
-        // Create temp files
-        try {
-            for (int i = 0; i < tempFilePaths.length; i++) {
-                tempFilePaths[i] = dir.resolve(tempFilesName[i]);
-                if (!Files.exists(tempFilePaths[i])){
-                    RandomAccessFile f = new RandomAccessFile(tempFilePaths[i].toFile(), "rw");
-                    f.setLength(tempFilesSize[i]);
-                    f.close();
+        F1_000_000(1_000_000, "1.0 MB"),
+        F1_000_001(1_000_001, "1.1 MB"),
+        F1_000_900(1_000_900, "1.1 MB"),
+        F1_001_000(1_001_000, "1.1 MB"),
+        F1_100_000(1_100_000, "1.1 MB"),
+        F1_100_001(1_100_001, "1.2 MB"),
+
+        F2_800_000(2_800_000, "2.8 MB"),
+
+//        F1_000_000_000(1_000_000_000, "1.0 GB"),
+//        F1_000_000_001(1_000_000_001, "1.1 GB"),
+        ;
+
+        public final String name;
+        public final long size;
+        public final String renderedSize;
+
+        private Path path;
+
+        FileSize(long size, String renderedSize) {
+            this.name = String.format("%03d-%010d.test", ordinal(), size);
+            this.size = size;
+            this.renderedSize = renderedSize;
+        }
+
+        public void create(final Path parent) {
+            path = parent.resolve(name);
+            if (!Files.exists(path)) {
+                try (var f = new RandomAccessFile(path.toFile(), "rw")) {
+                    f.setLength(size);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
                 }
             }
-            fc.setCurrentDirectory(dir.toFile());
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
         }
+
+        public void delete() {
+            try {
+                Files.deleteIfExists(path);
+            } catch (IOException e) {
+                e.printStackTrace();
+                // Don't propagate
+            }
+        }
+    }
+
+    private static JFrame frame;
+    private static JFileChooser fc;
+
+    private static final AtomicReference<String> error = new AtomicReference<>();
+
+    private static void createUI() {
+        // Create temp files
+        Path dir = Paths.get(".");
```

Comment:
It is better to create a testFolder, and then open the chooser in it, otherwise other files in the current folder may break it(if the test executed standalone)

Reply:
In my opinion, the current directory is a good enough option. When run in jtreg, it is set to `scratch` which is empty when the test starts; when run manually as a standalone app, the chances you have other files named `000-0000000000` in the current directory are quite low. The test does not depend on the first file being displayed in the first row of the table, it looks for it; yet it depends on all the files being displayed in the sequence without alternating with other files.